src/binheap.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8-*-
+
+
+class BinaryHeap(object):
+    """Python implementation of a Binary Heap Data Structure."""
+
+    def __init__(self, iter=None):
+        """Constructor function for BinaryHeap."""
+        self._heap_list = []
+
+        if iter:
+            for val in iter:
+                self.push(val)
+
+    @property
+    def heap_list(self):
+        return self._heap_list
+
+    def push(self, val):
+        """Put new value into the heap, maintaining the heap property."""
+        self._heap_list.append(val)
+        #import pdb; pdb.set_trace()
+        self._heap_check_up(self._parent_index(len(self._heap_list) - 1),
+                            len(self._heap_list) - 1)
+
+    def pop(self):
+        """Remove the top value in the heap, maintaining the heap property."""
+        self._heap_list.pop(0)
+        self._heap_check_down(self._heap_list[0])
+
+    def _heap_check_up(self, parent_index, child_index):
+        """Check if value of child is greater or less than parent.
+
+        Does a switch if this is true then does a check_up on the new parent
+        and its parent.
+        """
+        if child_index == 0 or parent_index is None:
+            return
+        if self._heap_list[child_index] > self._heap_list[parent_index]:
+            # Swtich the location of parent and child if child is greater.
+            self._heap_list[parent_index], self._heap_list[child_index] = \
```

Comment:
Avoid using the `\` line break at all costs. It can cause bugs when someone doesn't notice it when they're working on your code. Much better is using python's implicit line joining (https://google.github.io/styleguide/pyguide.html?showone=Line_length#Line_length)